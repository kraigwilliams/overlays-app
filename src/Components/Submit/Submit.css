.submit-form {
  padding-bottom: 20px;
  background: #f4d8cd;
  border: 1px solid white;
  border-radius: 200px;
  margin-left: auto;
  margin-right: auto;
  width: 50%;
  margin-top: 50px;
  display: flex;
  flex-direction: column;
  align-items: center;
  box-shadow: 4px 5px 7px 5px black;
}

.submit-form input {
  border-radius: 100px;
  border: 2px solid #1e555c;
  margin: 30px;
  width: 200px;
}

.submit-form label {
  color: #1e555c;
}

.submit-topic-btn {
  background: #f15152;
}
