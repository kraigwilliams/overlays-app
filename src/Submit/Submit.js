import React, { Component } from "react";

class Submit extends Component {
  onSubmit = event => {
      event.preventDefault()
      const {
          topic_name,
          topic_url,

      }= event.target
      fetch('http://localhost:8000/api/topics',{
        'method':'POST',
        headers:{
            'content-type': 'application/json'
        },
          body:JSON.stringify({
            topic_url:topic_url.value,
            topic_name:topic_name.value
          })
      })
      .then(res=>res.json())
      .then(data=>{
          console.log(data)
      })
  };

  render() {
    return (
      <>
        <form onSubmit={this.onSubmit}>
          <input type="text" name="topic_url" placeholder="Url to Add" />
          <br />
          <br />
          <input type="text" name="topic_name"placeholder="Topic Name " />
          <br />
          <br />
          <input type="submit" value="submit" />
        </form>
      </>
    );
  }
}

export default Submit;
