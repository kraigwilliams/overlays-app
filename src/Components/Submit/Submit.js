import React, { Component} from "react";
import config from '../../config'
//import topicsApiService from '../../services/topics-api-service';
import TokenService from '../../services/token-service';
import './Submit.css'
import {Button,Input}from '../Utils/Utils';
class Submit extends Component {

  //  state={
  //   topics:[]
  // }

  componentDidMount(){
    document.title = "Submit New Url Topic - Overlays";
  }

  
  onSubmit = event => {
    event.preventDefault();
  
    const { topic_name, topic_url } = event.target;
    fetch(`${config.API_ENDPOINT}/topics`,{
    
      method: "POST",
      headers: {
        "content-type": "application/json",
        authorization:"bearer " + TokenService.getAuthToken()
      },
      body: JSON.stringify({
        topic_url: topic_url.value,
        topic_name: topic_name.value
      })
    })
      //.then(res => res.json())
      .then( 
        
        this.props.history.push('/topics')
      
      );
  };

  render() {
    return (
      <>
      <h2>Submit New Topic</h2>
        <form className="submit-form" onSubmit={this.onSubmit}>
          <h3><label htmlFor="submit-name">New Topic Name </label></h3>
        <Input id="submit-name" type="text" name="topic_name" placeholder="Submit New Topic" />
          
        
          <h3><label htmlFor="submit-url">New Topic URL</label></h3>
          <Input id="submit-url" type="text" name="topic_url" placeholder="Url to Add" />
          
          <Button className="submit-topic-btn" type="submit"> Submit</Button>
        </form>
      </>
    );
  }
}

export default Submit;
