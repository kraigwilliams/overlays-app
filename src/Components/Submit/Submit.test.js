import React from 'react';
//import { render } from '@testing-library/react';
import Submit from './Submit'
import {BrowserRouter} from 'react-router-dom'
import ReactDOM from  'react-dom';
import renderer from 'react-test-renderer';

describe('<Submit/>',()=>{
  it ('is able to render',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><Submit/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><Submit /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

  
})