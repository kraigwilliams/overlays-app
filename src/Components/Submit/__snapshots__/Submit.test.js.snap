// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Submit/> renders the UI as expected 1`] = `
Array [
  <h2>
    Submit New Topic
  </h2>,
  <form
    className="submit-form"
    onSubmit={[Function]}
  >
    <h3>
      <label
        htmlFor="submit-name"
      >
        New Topic Name 
      </label>
    </h3>
    <input
      className="Input "
      id="submit-name"
      name="topic_name"
      placeholder="Submit New Topic"
      type="text"
    />
    <h3>
      <label
        htmlFor="submit-url"
      >
        New Topic URL
      </label>
    </h3>
    <input
      className="Input "
      id="submit-url"
      name="topic_url"
      placeholder="Url to Add"
      type="text"
    />
    <button
      className="Button submit-topic-btn"
      type="submit"
    >
       Submit
    </button>
  </form>,
]
`;
